fastdialer/dialer.go

```diff
@@ -229,7 +232,29 @@ func (d *Dialer) dial(ctx context.Context, network, address string, shouldUseTLS
 			}
 			conn, err = ztls.DialWithDialer(d.dialer, network, hostPort, ztlsconfigCopy)
 		} else {
-			conn, err = d.dialer.DialContext(ctx, network, hostPort)
+			if d.proxyDialer != nil {
+				dialer := *d.proxyDialer
+				// timeout not working for socks5 proxy dialer
+				// tying to handle it here
+				connectionCh := make(chan net.Conn, 1)
+				errCh := make(chan error, 1)
+				go func() {
+					conn, err = dialer.Dial(network, hostPort)
+					if err != nil {
+						errCh <- err
+						return
+					}
+					connectionCh <- conn
+				}()
+				select {
+				case <-time.After(d.options.DialerTimeout):
```

Comment:
`time.After` leaks go routines as it's not recovered gy `GC` immediately. Probably here we can use:
```go
...
// using timer as time.After is not recovered gy GC
dialerTime := time.NewTimer(d.options.DialerTimeout)
defer dialerTime.Stop()
...
case <-dialerTime.C:
...
```